include/boost/geometry/algorithms/detail/overlay/traversal.hpp

```diff
@@ -124,7 +106,7 @@ private :
 
     static const operation_type target_operation = operation_from_overlay<OverlayType>::value;
 
-    typedef typename side_compare<target_operation>::type side_compare_type;
+    typedef typename sort_by_side::side_compare<target_operation>::type side_compare_type;
```

Comment:
This is not a change request but a question. I'm wondering, since this is the only place where `side_compare` is used, is there a reason why separate struct template was added and not MPL/TT used to define this member type instead? E.g.:

    typedef typename boost::mpl::if_c
        <
            boost::is_same<target_operation, operation_union>::value,
            std::greater<int>,
            std::less<int>
        >::type side_compare_type;

Reply:
That would also be possible. This was written years ago when we didn't use is_same so often.
Besides that, now you will get a compilation error if it is not either union or intersection. With the same, you won't get that. So it might be more error-prone (depending what you want, of course)